src/main/scala/DataBaseConn.scala

```diff
@@ -1,14 +1,67 @@
 import DataBaseConn.xa
-import cats.effect.IO
+import cats.effect.{Blocker, IO}
+import com.zaxxer.hikari.HikariDataSource
 import doobie._
+import doobie.hikari.HikariTransactor
 import doobie.implicits._
+import doobie.implicits.javasql._
+
+import java.sql.Timestamp
 import scala.concurrent.ExecutionContext
 
+trait UrlDatabaseFunctions {
+  def storeUrl(userId: Int, timestamp: Timestamp, value: String): Int
+  def getAllUrls: List[Url]
+  def getAllUrlsForUser(userId: Int): List[Url]
+  def getUrlById(urlId: Int): Option[Url]
+  def getUrlByUserIdAndUrlId(urlId: Int): Option[Url]
+}
+
+object UrlDatabaseImplementation extends UrlDatabaseFunctions {
+
+  def storeUrl(userId: Int, timestamp: Timestamp, value: String): Int = {
+    sql"insert into urls(timestamp, value, userid) values ($timestamp, $value, $userId)"
+      .update
+      .run
+      .transact(xa)
+      .unsafeRunSync()
+  }
+
+  def getAllUrls: List[Url] = {
+    sql"select userid, value, timestamp from urls".query[Url]
+      .to[List]
+      .transact(xa)
+      .unsafeRunSync()
+  }
+
+  def getAllUrlsForUser(userId: Int): List[Url] = {
+    sql"select userid, value, timestamp from urls where userid=$userId".query[Url]
+      .to[List]
+      .transact(xa)
+      .unsafeRunSync()
+  }
+
+  def getUrlById(urlId: Int): Option[Url] = {
+    sql"select userid, value, timestamp from urls where id=$urlId".query[Url]
```

Comment:
You're missing some formatting here
```suggestion
    sql"select userid, value, timestamp from urls where id = $urlId".query[Url]
```